core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainer.scala

```diff
@@ -236,7 +236,7 @@ class DockerContainer(protected val id: ContainerId,
   def logs(limit: ByteSize, waitForSentinel: Boolean)(implicit transid: TransactionId): Source[ByteString, Any] = {
     docker
       .rawContainerLogs(id, logFileOffset.get(), if (waitForSentinel) Some(filePollInterval) else None)
-      .via(Framing.delimiter(delimiter, Int.MaxValue))
+      .via(Framing.delimiter(delimiter, limit.toBytes.toInt))
```

Comment:
Should add a comment here that we can obtain a FramingException for two reasons:
1. the Source provides an incomplete line so that the delimiter cannot be found.
2. the Source provides more data than `limit.toBytes.toInt` which is effectively a truncation due to limit exceeded.

As a result we cannot decide in later stages whether we should issue a `log processing problem` or a `log truncated` message.